WORKSPACE

```diff
@@ -54,23 +53,106 @@ yarn_install(
+    },
+    package_json = "//internal/linker/test/multi_linker/test_a:package.json",
+    package_path = "internal/linker/test/multi_linker/test_a",
+    symlink_node_modules = False,
+    yarn_lock = "//internal/linker/test/multi_linker/test_a:yarn.lock",
+)
+
+yarn_install(
+    name = "internal_test_multi_linker_test_b_deps",
+    package_json = "//internal/linker/test/multi_linker/test_b:package.json",
+    package_path = "internal/linker/test/multi_linker/test_b",
     symlink_node_modules = False,
-    yarn_lock = "//internal/linker/test/multi_linker/onepa:yarn.lock",
+    yarn_lock = "//internal/linker/test/multi_linker/test_b:yarn.lock",
+)
+
+yarn_install(
+    name = "internal_test_multi_linker_test_c_deps",
+    package_json = "//internal/linker/test/multi_linker/test_c:package.json",
+    package_path = "internal/linker/test/multi_linker/test_c",
+    symlink_node_modules = False,
+    yarn_lock = "//internal/linker/test/multi_linker/test_c:yarn.lock",
+)
+
+yarn_install(
+    name = "internal_test_multi_linker_test_d_deps",
+    package_json = "//internal/linker/test/multi_linker/test_d:package.json",
+    package_path = "internal/linker/test/multi_linker/test_d",
+    symlink_node_modules = False,
+    yarn_lock = "//internal/linker/test/multi_linker/test_d:yarn.lock",
+)
+
+yarn_install(
+    name = "internal_test_multi_linker_lib_b_deps",
+    # transitive deps for this first party lib should not include dev dependencies
+    args = ["--production"],
+    package_json = "//internal/linker/test/multi_linker/lib_b:package.json",
+    package_path = "internal/linker/test/multi_linker/lib_b",
+    symlink_node_modules = False,
+    yarn_lock = "//internal/linker/test/multi_linker/lib_b:yarn.lock",
+)
+
+yarn_install(
+    name = "internal_test_multi_linker_sub_dev_deps",
+    local_deps = {
+        "@test_multi_linker/lib-a": "@build_bazel_rules_nodejs//internal/linker/test/multi_linker/lib_a",
+        "@test_multi_linker/lib-b": "@build_bazel_rules_nodejs//internal/linker/test/multi_linker/lib_b",
+    },
+    package_json = "//internal/linker/test/multi_linker/sub:package.json",
+    package_path = "internal/linker/test/multi_linker/sub",
+    symlink_node_modules = False,
+    yarn_lock = "//internal/linker/test/multi_linker/sub:yarn.lock",
+)
+
+yarn_install(
+    name = "internal_test_multi_linker_sub_deps",
+    # transitive deps for this first party lib should not include dev dependencies
+    args = ["--production"],
+    local_deps = {
+        "@test_multi_linker/lib-a": "@build_bazel_rules_nodejs//internal/linker/test/multi_linker/lib_a",
+        "@test_multi_linker/lib-b": "@build_bazel_rules_nodejs//internal/linker/test/multi_linker/lib_b",
+    },
+    package_json = "//internal/linker/test/multi_linker/sub:package.json",
+    package_path = "internal/linker/test/multi_linker/sub",
+    symlink_node_modules = False,
+    yarn_lock = "//internal/linker/test/multi_linker/sub:yarn.lock",
+)
+
+yarn_install(
```

Comment:
Cross my mind as well. I have it as a post-land cleanup since I'll move all of the existing yarn_installs + the ones added here. Could be a pre-factor instead.